src/components/subscriptions/addons/edit/AddonRatesEditor.js

```diff
@@ -0,0 +1,127 @@
+                    required
+                    validate={(value) => minValue(value, i18nUtil)}
+                />
+            </TableCell>
+            <TableCell>
+                <Field
+                    component={FormTimestampField}
+                    id={buildID(
+                        baseId,
+                        ids.ADDONS_DLG.ADDON_RATE.EFFECTIVE_DATE
+                    )}
+                    name={`${fieldName}.effectiveDate`}
+                    required
+                    validate={(value) => nonEmptyField(value, i18nUtil)}
+                />
+            </TableCell>
+            <TableCell passing="none">
+                <Button
+                    id={buildID(baseId, ids.DELETE_BUTTON)}
+                    aria-label={t("common:delete")}
+                    onClick={() => {
+                        onDelete(key);
+                    }}
+                >
+                    <Delete />
+                </Button>
+            </TableCell>
+        </TableRow>
+    );
+}
+
+function AddonRatesEditor(props) {
+    const { addonRates, baseId, fieldName, onAdd, onDelete } = props;
+
+    const { t } = useTranslation(["subscriptions", "common"]);
+
+    return (
+        <>
+            <TableContainer>
+                <Table>
+                    <Typography component="caption">
+                        {t("subscriptions:addonRates")}
+                    </Typography>
+                    <TableHead>
+                        <TableRow>
+                            <TableCell>
+                                <Typography>
+                                    {t("subscriptions:rate")}
+                                </Typography>
+                            </TableCell>
+                            <TableCell>
+                                <Typography>
+                                    {t("subscriptions:effectiveDate")}
+                                </Typography>
+                            </TableCell>
+                            <TableCell>
+                                <Button
+                                    id={buildID(baseId, ids.ADD_BUTTON)}
+                                    color="primary"
+                                    variant="outlined"
+                                    startIcon={<Add />}
+                                    onClick={onAdd}
+                                >
+                                    {t("common:add")}
+                                </Button>
+                            </TableCell>
+                        </TableRow>
+                    </TableHead>
+                    <TableBody>
+                        {addonRates?.map((addonRate, index) => (
+                            <AddonRateEditorRow
+                                addonRate={addonRate}
+                                baseId={buildID(baseId, index)}
+                                fieldName={`${fieldName}.${index}`}
+                                key={index}
```

Comment:
It's possible using the `index` as a `key` may cause issues when addon rates get deleted in the form (though it should be tested since it might only cause issues when rates are deleted and inserted, or if they need to be rearranged in the future).

It might be better to use `addonRate.uuid` as the `key`, but then a new `uuid` would have to be created in the `onAdd` function with `crypto.randomUUID()`.

Reply:
Using the UUID as the key makes more sense to me than using the index. I'll update it. Thanks for the suggestion!